caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -201,6 +205,29 @@ def public_model_info(cls) -> Dict[str, Any]:  # pylint: disable=no-self-argumen
             "sentence_embedding_dimension": cls.model.get_sentence_embedding_dimension(),
         }
 
+    @TokenizationTask.taskmethod()
+    def run_tokenizer(
+        self,
+        text: str,
+    ) -> TokenizationResults:
+        """Run tokenization task against the model
+
+        Args:
+            text: str
+                Text to tokenize
+        Returns:
+            TokenizationResults
+                The token count
+        """
+        result = self.model._get_tokenized([text])
+
+        mapping = [
+            interv for interv in result.offset_mapping[0] if (interv[1] - interv[0]) > 0
+        ]
+        tokens = [Token(start=i[0], end=i[1], text=text[i[0] : i[1]]) for i in mapping]
+
+        return TokenizationResults(token_count=len(result.input_ids), results=tokens)
```

Comment:
This needs index zero because we wrapped the text in a list for _get_tokenized().

```suggestion
        return TokenizationResults(token_count=len(result.input_ids[0), results=tokens)
```